test/test_library/unit/test_init.py

```diff
@@ -0,0 +1,1858 @@
+
+SQLITE_DATABASE = 'test_library_test_init.db'
+BUNDLES_DIR_PREFIX = 'test_library_test_init_bundles'
+
+
+class NewLibraryTest(unittest.TestCase):
+    def setUp(self):
+        self.sqlite_db = LibraryDb(driver='sqlite', dbname=SQLITE_DATABASE)
+        self.sqlite_db.enable_delete = True
+        self.sqlite_db.create_tables()
+
+    def tearDown(self):
+        self.sqlite_db.close()
+        try:
+            os.remove(SQLITE_DATABASE)
+        except OSError:
+            pass
+
+    def test_raises_DatabaseError_if_database_creation_failed(self):
+
+        fake_create = fudge.Fake().expects_call().raises(OperationalError('select 1;', [], 'a'))
+        config = {
+            'database': {
+                'driver': 'sqlite',
+                'dbname': 'unused1.db',
+            },
+            'filesystem': 'http://example.com'}
+        with fudge.patched_context(LibraryDb, 'create', fake_create):
+            with self.assertRaises(DatabaseError):
+                _new_library(config)
+
+    def test_upstream_setting_is_deprecated(self):
+
+        config = {
+            'database': {
+                'driver': 'sqlite',
+                'dbname': 'test_init1_unused.db',
+            },
+            'filesystem': 'http://example.com',
+            'upstream': 'upstream'}
+        with self.assertRaises(DeprecationWarning):
+            # prevent sqlite db file creation.
+            with fudge.patched_context(LibraryDb, 'create', fudge.Fake().is_a_stub()):
+                _new_library(config)
+
+    def test_logs_S3ResponseError(self):
+        fake_s3 = fudge.Fake('s3').has_attr(bucket='bucket1')
+
+        config = {
+            'database': {
+                'driver': 'sqlite',
+                'dbname': 'test_init1_unused.db',
+            },
+            'filesystem': 'http://example.com',
+            'remotes': {'s3': fake_s3},
+            'root': 'UNUSED?'}
+
+        fake_new_cache = fudge.Fake()\
+            .expects_call().returns(fudge.Fake().is_a_stub())\
+            .next_call().raises(S3ResponseError('status', 'reason'))
+
+        fake_logger = fudge.Fake('logger')\
+            .expects('error').with_args(arg.contains('Failed to init'))
+
+        # get_logger need 3 return values, first for the LibraryDb constructor logger, second
+        # for the _new_library logger, third for Library constructor.
+        fake_get_logger = fudge.Fake('get_logger')\
+            .expects_call().returns(fudge.Fake().is_a_stub())\
+            .next_call().returns(fake_logger)\
+            .next_call().returns(fudge.Fake().is_a_stub())
+
+        with fudge.patched_context(ckcache, 'new_cache', fake_new_cache):
+            with fudge.patched_context(util, 'get_logger', fake_get_logger):
+                # prevent sqlite db file creation.
+                with fudge.patched_context(LibraryDb, 'create', fudge.Fake().is_a_stub()):
+                    _new_library(config)
+
+    @unittest.skip('host and port are never used.')
```

Comment:
Remove host and port processing from here - https://github.com/nmb10/ambry/blob/issue-8/ambry/library/__init__.py#L69